doc/releases/changelog-0.41.0.md

```diff
@@ -691,6 +729,43 @@ With `qml.decompositions.enable_graph()`, the following new features are availab
   [(#6938)](https://github.com/PennyLaneAI/pennylane/pull/6938)
   [(#7037)](https://github.com/PennyLaneAI/pennylane/pull/7037)
 
+<h4>Program capture and dynamic shapes</h4>
```

Comment:
FYI I moved these features from the "New features" section into their own program capture section in improvements